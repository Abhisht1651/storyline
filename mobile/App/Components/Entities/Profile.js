import React from 'react'
import { View, Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from '../Styles/EntitesStyles/ProfileStyles'
import images from '../../Themes/Images'

const Profile = ({ profileIcons }) => {
  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image source={profileIcon} key={index} style={styles.profileIcon} />
  ))
  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>
    </View>
  )
}

Profile.propTypes = {
  profileIcons: PropTypes.array
}

Profile.defaultProps = {
  profileIcons: [images.EntitiesImage0]
}

export default Profile
