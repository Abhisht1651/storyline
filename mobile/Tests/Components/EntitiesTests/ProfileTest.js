import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import Profile from '../../../App/Components/Entities/Profile'

test('CardType1 Entity component renders correctly', () => {
  const tree = renderer.create(<Profile />).toJSON()
  expect(tree).toMatchSnapshot()
})
