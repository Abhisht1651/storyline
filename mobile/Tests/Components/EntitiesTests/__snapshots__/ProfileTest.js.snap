// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardType1 Entity component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 30,
      "height": 173.69791666666669,
      "width": 200,
    }
  }
>
  <View
    style={
      Object {
        "height": 173.69791666666669,
        "width": 200,
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "borderRadius": 30,
          "height": 173.69791666666669,
          "width": 200,
        }
      }
    />
  </View>
</View>
`;
